src/vs/workbench/services/views/browser/viewsService.ts

```diff
@@ -664,28 +662,37 @@ export class ViewsService extends Disposable implements IViewsService {
 		Registry.as<PaneCompositeRegistry>(getPaneCompositeExtension(viewContainerLocation)).deregisterPaneComposite(viewContainer.id);
 	}
 
-	private createViewPaneContainer(element: HTMLElement, viewContainer: ViewContainer, viewContainerLocation: ViewContainerLocation, disposables: DisposableStore, instantiationService: IInstantiationService): ViewPaneContainer {
+	private createViewPaneContainer(viewContainer: ViewContainer, disposables: DisposableStore, instantiationService: IInstantiationService): ViewPaneContainer {
 		const viewPaneContainer: ViewPaneContainer = (instantiationService as any).createInstance(viewContainer.ctorDescriptor.ctor, ...(viewContainer.ctorDescriptor.staticArguments || []));
 
 		this.viewPaneContainers.set(viewPaneContainer.getId(), viewPaneContainer);
 		disposables.add(toDisposable(() => this.viewPaneContainers.delete(viewPaneContainer.getId())));
 		disposables.add(viewPaneContainer.onDidAddViews(views => this.onViewsAdded(views)));
-		disposables.add(viewPaneContainer.onDidChangeViewVisibility(view => this.onViewsVisibilityChanged(view, view.isBodyVisible())));
+		disposables.add(viewPaneContainer.onDidChangeViewVisibility(view => {
+			this.onViewsVisibilityChanged(view, view.isBodyVisible());
+			if (!view.isVisible()) {
```

Comment:
@benibenj this is maybe up for debate: there is 2 visibilities for views, the body and the entire view. For the `onViewsVisibilityChanged` we seem to be using the body visibility. For the focused view I am using the entire view visibility to cover the issue reported by Ulugbek.

Reply:
Yeah I tend to agree, and we already had a hack for this here:

https://github.com/microsoft/vscode/blob/997fd30cc5df08a40fc255be76051720a2fc9f78/src/vs/workbench/services/views/browser/viewsService.ts#L337-L342